lib/jenkins_api_client/client.rb

```diff
@@ -602,14 +602,25 @@ def get_hudson_version
     # Converts a version string to a list of integers
     # This makes it easier to compare versions since in 'version-speak',
     # v 1.2 is a lot older than v 1.102 - and simple < > on version
-    # strings doesn't work so well
+    # strings doesn't work so well. There could be an indefinite
+    # number of decimals, but for this gem we only care about the first four.
+    # If there are alpha characters, return nil.
     def deconstruct_version_string(version)
-      match = version.match(/^(\d+)\.(\d+)(?:\.(\d+))?$/)
-
-      # Match should have 4 parts [0] = input string, [1] = major
-      # [2] = minor, [3] = patch (possibly blank)
-      if match && match.size == 4
-        return [match[1].to_i, match[2].to_i, match[3].to_i || 0]
+      
+      vers_array = []
+      matches = version.match(/[a-zA-Z]/)
+      return nil if matches
+      version.scan(/\d+/) { |ver|
+        vers_array.push(ver.to_i)
+      }
+      if vers_array[2].nil?
```

Comment:
Since `nil.to_i` returns 0, we can reduce these two conditionals to:
```
     vers_array[2] = vers_array[2].to_i
     vers_array[3] = vers_array[3].to_i
```